drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -87,14 +87,21 @@ local function convert_huesat_st_to_matter(val)
   return utils.clamp_value(math.floor((val * 0xFE) / 100.0 + 0.5), CURRENT_HUESAT_ATTR_MIN, CURRENT_HUESAT_ATTR_MAX)
 end
 
-local function mired_to_kelvin(value)
+local function mired_to_kelvin(value, minOrMax)
   if value == 0 then -- shouldn't happen, but has
     value = 1
     log.warn(string.format("Received a color temperature of 0 mireds. Using a color temperature of 1 mired to avoid divide by zero"))
   end
-  -- we divide inside the rounding and multiply outside of it because we expect these
-  -- bounds to be multiples of 100
-  return utils.round((MIRED_KELVIN_CONVERSION_CONSTANT / value) / 100) * 100
+  -- We divide inside the rounding and multiply outside of it because we expect these
+  -- bounds to be multiples of 100. For the maximum mired value (minimum K value),
+  -- add 1 before converting and round up to nearest hundreds. For the minimum mired
+  -- (maximum K value) value, subtract 1 before converting and round down to nearest
+  -- hundreds.
+  if minOrMax == COLOR_TEMP_MIN then
+    return utils.round(((MIRED_KELVIN_CONVERSION_CONSTANT / (value + 1)) / 100) + 0.5) * 100
+  else
```

Comment:
```suggestion
  elseif minOrMax == COLOR_TEMP_MAX then
  ...
  else
  -- add an error message here
  end
```
I think this makes it more clear what the expected values are.

Reply:
Updated as suggested in the latest commit!